src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -469,80 +469,87 @@ void QgsRelationEditorWidget::linkFeature()
   else
     layer = mRelation.referencingLayer();
 
-  QgsFeatureSelectionDlg selectionDlg( layer, mEditorContext, this );
+  QgsFeatureSelectionDlg *selectionDlg = new QgsFeatureSelectionDlg( layer, mEditorContext, this );
```

Comment:
Is this leaked?

Reply:
Yes, I added the deleteonclose attribute

Reply:
Are we sure this works on hiding the dialog (or that the dialog is always closed and never hidden)?
Sorry for bumping this question, I'm not sure at all, I just want to be sure the DeleteOnClose concept doesn't have some hidden caveats.

Reply:
Tested it on debug mode, dialog is well deleted when you close it, no matter how (cancel, accept, close icon). The dialog is always closed, never hidden. 

I think the only way to hide a dialog is programmatically calling hide() ou setVisible(False) method. In this case, I assume the dialog won't be deleted.